storage/innobase/row/row0merge.cc

```diff
@@ -4119,7 +4120,15 @@ row_merge_file_create_low(
 		PSI_FILE_CREATE, path ? name : label, __FILE__, __LINE__);
 
 #endif
-	pfs_os_file_t fd = innobase_mysql_tmpfile(path);
+	{
+		os_file_t f = innobase_mysql_tmpfile(path);
+#if defined _WIN32
+		fd = my_get_osfhandle(f);
+#else
+		fd = f;
+#endif
+		my_invalidate_fd(f);
```

Comment:
```c
fd = IF_WIN(my_get_osfhandle(f), f);
```
would be nicer to read.
@svoj Do you agree that calling `my_invalidate_fd(f)` is OK here? I am a little uncertain how the ownership is being transferred here.
One more thing: will `my_get_osfhandle(-1)` return a value that is equivalent to `OS_FILE_CLOSED`? If not, then we should add special handling for `f<0`.

Reply:
`my_invalidate_fd` is a release of ownership. Its one of the things that can get pulled out when native windows file handles come into mysys. `-1` being the same as `OS_FILE_CLOSED` was covered in the review last year.